deploy/examples/multus-validation.yaml

```
@@ -0,0 +1,138 @@
+# allowHostPorts: true
+# priority:
+# allowedCapabilities: ["MKNOD"]
+# allowHostIPC: true
+# readOnlyRootFilesystem: false
+# # drop all default privileges
+# requiredDropCapabilities: ["All"]
+# defaultAddCapabilities: []
+# runAsUser:
+#   type: RunAsAny
+# seLinuxContext:
+#   type: RunAsAny
+# fsGroup:
+#   type: RunAsAny
+# supplementalGroups:
+#   type: RunAsAny
+# seccompProfiles:
+#   - "*"
+# volumes:
+#   - configMap
+#   - emptyDir
+#   - projected
+# users:
+#   - system:serviceaccount:rook-ceph:rook-ceph-multus-validation # serviceaccount:namespace:cluster
+---
+# Aspects of multus validation job that require access to the operator/cluster namespace
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: rook-ceph-multus-validation
+  namespace: rook-ceph # namespace:cluster
+rules:
+  - apiGroups: [""]
+    resources: ["configmaps", "configmaps/finalizers", "pods"]
+    verbs: ["get", "list", "create", "update", "delete"]
+  - apiGroups: ["apps"]
+    resources: ["daemonsets"]
+    verbs: ["list", "create", "delete", "deletecollection"]
+  - apiGroups: ["k8s.cni.cncf.io"]
+    resources: ["network-attachment-definitions"]
+    verbs: ["get"]
+  - apiGroups: ["batch"]
+    resources: ["jobs"]
+    verbs: ["get", "list", "delete"]
+---
+# Allow the multus validation job to run in this namespace
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: rook-ceph-multus-validation
+  namespace: rook-ceph # namespace:cluster
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: rook-ceph-multus-validation
+subjects:
+  - kind: ServiceAccount
+    name: rook-ceph-multus-validation
+    namespace: rook-ceph # namespace:cluster
+---
+# Service account for job that validates multus configuration
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: rook-ceph-multus-validation
+  namespace: rook-ceph # namespace:cluster
+# imagePullSecrets:
+#   - name: my-registry-secret
+---
+# A job that runs the multus validation tool
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: rook-ceph-multus-validation
+  namespace: rook-ceph # namespace:cluster
+  labels:
+    app: rook-ceph-multus-validation
+spec:
+  template:
+    metadata:
+      labels:
+        app: rook-ceph-multus-validation
+    spec:
+      serviceAccountName: rook-ceph-multus-validation
+      containers:
+        - name: multus-validation
+          image: rook/ceph:master
+          command: ["rook"]
+          # TODO: Insert the NAD name for public network and cluster network
+          # If you want to use any other flags along with the basic command,
+          # add the `--help` flag in the end to see the list of flags available, and use accordingly.
+          args:
+            - "multus"
+            - "validation"
+            - "run"
+            - "--public-network"
+            - "<nad-name>"
+            - "--cluster-network"
+            - "<nad-name>"
```

Let's comment these out by default so that anyone running this without reading it won't have the tool seem to run with environment/test failures, and instead it will be failures from not providing input.

```suggestion
            # - --public-network=<NAD-NAME> # uncomment and replace NAD name if using public network
            # - --cluster-network=<NAD-NAME> # uncomment and replace NAD name if using cluster network
```